### PR TITLE
allow multiline string comments

### PR DESCRIPTION
**Pull request by mpollmeier**

### CODE REVIEW

**Review by bbrehm — CHANGES_REQUESTED** [review]
While we're at it, we should just escape `doc`. Otherwise we're going to be unhappy if we ever need comments that contain backslashes, `*/`, triple-quotes, etc.

`org.apache.commons.lang.StringEscapeUtils.escapeJava` is probably appropriate for the job.

**Comment by mpollmeier** [comment]
Good idea, however I'd prefer to treat this as a separate issue and not put a stop sign on this PR. 

To your suggestion:
`org.apache.commons.lang.StringEscapeUtils.escapeJava` is deprecated and replaced by `org.apache.commons.text.StringEscapeUtils.escapeJava`
(`"org.apache.commons" % "commons-text" % "1.9"`)
https://commons.apache.org/proper/commons-text/javadocs/api-release/org/apache/commons/text/StringEscapeUtils.html

AFAICS it doesn't really help us, because it escapes things we don't need to escape (e.g. newline, tab etc.), and doesn't escape things we do need (e.g. `/*`, `*/` etc.). I just added a custom escape function - please let me know if I missed any escapeworthy strings.

**Review by bbrehm — APPROVED** [review]
Ok, I don't want to stop other progress and it's not critical to get the escaping really airtight.

...but I think we need different escaping? That is: `info = \"${escapeJava(doc)}\"` and `$indent/** ${escapeComment($doc)}*/`. `escapeJava` needs to escape a string for use in java string literals (e.g. escape double-quotes), while escapeComment needs to escape the string for use in comments (e.g. escape `/*`).

Something that needs to be always escaped is patterns of the style `\u1234` -- cf https://docs.oracle.com/javase/specs/jls/se8/html/jls-3.html 

Per java spec (and real-life javac and scalac behavior) and opposed to the intellij parser, unicode escapes are resolved before tokenization / lexing.

So the following would blow up: `comment = "doc\u005c\u005cu002f*"`. javac would transform this into `comment = "doc\\u00ef*"` before parsing while compiling codegen; then, when codegen executes,  we get `/** doc\u00ef */` in the resulting generated source-file; and finally, javac would transform that into `/** doc /* */` before parsing the generated source file -- and then the comment doesn't end where you think it ends.

Btw, intellij and vscode dont handle unicode-escapes that escape a comment correctly either, so this is not entirely trivial to spot because the syntax highlighting is lying to you. Entry for the obfuscated java contest, here we come.

TLDR: correctly escaping strings for inclusion in java source files is surprisingly annoying. Let a library deal with it?
